dataland-backend/src/main/kotlin/org/dataland/datalandbackend/utils/DataPointValidator.kt

```diff
@@ -0,0 +1,127 @@
+import org.dataland.specificationservice.openApiClient.api.SpecificationControllerApi
+import org.dataland.specificationservice.openApiClient.infrastructure.ClientException
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Class for validating data points
+ * @param objectMapper the object mapper to use for JSON operations
+ * @param specificationClient the specification client to use for fetching specifications
+ */
+
+@Component("DataPointValidator")
+class DataPointValidator(
+    @Autowired private val objectMapper: ObjectMapper,
+    @Autowired private val specificationClient: SpecificationControllerApi,
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Validates a single data point by casting it to the correct class and running the validations
+     * @param dataPointIdentifier the identifier of the data point
+     * @param dataPointContent the content of the data point
+     * @param correlationId the correlation id for the operation
+     */
+    fun validateDataPoint(
+        dataPointIdentifier: String,
+        dataPointContent: String,
+        correlationId: String,
+    ) {
+        logger.info("Validating data point $dataPointIdentifier (correlation ID: $correlationId)")
+        validateDataPointIdentifierExists(dataPointIdentifier)
+        val dataPointType = specificationClient.getDataPointSpecification(dataPointIdentifier).validatedBy.id
+        val validationClass = specificationClient.getDataPointTypeSpecification(dataPointType).validatedBy
+        validateConsistency(dataPointContent, validationClass, correlationId)
+    }
+
+    /**
+     * Validates with the specification service that a data point identifier exists
+     * @param dataPointIdentifier the identifier to validate
+     */
+    fun validateDataPointIdentifierExists(dataPointIdentifier: String) {
+        try {
+            specificationClient.getDataPointSpecification(dataPointIdentifier)
+        } catch (clientException: ClientException) {
+            logger.error("Data point identifier $dataPointIdentifier not found: ${clientException.message}.")
+            throw InvalidInputApiException(
+                "Specified data point identifier $dataPointIdentifier is not valid.",
+                "The specified data point identifier $dataPointIdentifier is not known to the specification service.",
+            )
+        }
+    }
+
+    /**
+     * Validates the consistency of a JSON string with a given class.
+     * @param jsonData The JSON string to validate
+     * @param className The name of the class to validate against
+     * @param correlationId The correlation ID of the operation
+     */
+    fun validateConsistency(
+        jsonData: String,
+        className: String,
+        correlationId: String,
+    ) {
+        val classForValidation = Class.forName(className).kotlin.java
```

Comment:
My IT-Security training does not like this function. If somebody were to ever get control of the `className` field, he could likely trigger some kind of remote-code-execution. What do you think about adding a filter and only allowing `className` starting with `org.dataland.datalandbackend.model.datapoints.`?

Reply:
done